crypto/blake2/blake2b.c

```diff
@@ -81,6 +89,11 @@ static void blake2b_init_param(BLAKE2B_CTX *S, const BLAKE2B_PARAM *P)
 /* Initialize the hashing context.  Always returns 1. */
```

Comment:
I hate this comment.

We're now returning a zero but it seems that the return is expected a qualified structure that may be just randomly allocated space and allow the other functions like `blake2b_compress` to run unexpectedly.

We might need to add the fips check to `static void black2b_compress`
```
    if (FIPS_mode()) {
        FIPSerr(ERR_LIB_FIPS, FIPS_R_NON_FIPS_METHOD);
        return 0;
    }
```

Reply:
Well it's not my comment :-). 

Also, the definition of blake2b_compress() is:

static void blake2b_compress()

which means there's no way to get to this externally (it's defined static). So I don't think the fips check is needed in there.

Reply:
Yeah its called at the end of this `init` function.

I missed the static part of the compress when I saw that the fips check was missing and the weird statement in the comment that the `inti` will always return 1 then saw that `BLAKE2B_CTX *c` was memset with in `int BLAKE2b_Init(BLAKE2B_CTX *c)` that I then forgot the start of the fuction and jumped into how it could have been used in that function.

Reply:
Gonna resolve